extension-jaxrs/src/main/java/io/smallrye/openapi/jaxrs/JaxRsAnnotationScanner.java

```diff
@@ -216,7 +216,10 @@ private OpenAPI processApplicationClass(final AnnotationScannerContext context,
 
     private void processResourceClasses(final AnnotationScannerContext context, OpenAPI openApi) {
         // Now find all jax-rs endpoints
-        Collection<ClassInfo> resourceClasses = getJaxRsResourceClasses(context);
+        Collection<ClassInfo> resourceClasses = new ArrayList<>();
+        resourceClasses.addAll(getJaxRsResourceClasses(context));
+        resourceClasses.addAll(getConfigurationResourceClasses(context));
```

Comment:
In addition to this, once the class is being processed, you'll need to again look at the config's `getScanResourceClasses` to determine the path (since there is no `@Path`)

Reply:
This is exactly what I didn't figure out how to do.
As I understand in `JaxRsParameterProcessor` we have to create a parameter from the path specified in the configuration, so I tried something like this, but without success:
```
    @Override
    protected void processPathParameters(ClassInfo resourceClass, MethodInfo resourceMethod, ResourceParameters parameters) {
        ClassInfo resourceMethodClass = resourceMethod.declaringClass();

        // This is what I added
        readConfigurationPathParameter(resourceClass);
        
        /*
         * Phase I - Read class fields, constructors, "setter" methods not annotated with JAX-RS
         * HTTP method. Check both the class declaring the method as well as the resource
         * class, if different.
         */
        readParametersInherited(resourceMethodClass, null, false);

        if (!resourceClass.equals(resourceMethodClass)) {
            /*
             * The resource class may be a subclass/implementor of the resource method class. Scanning
             * the resource class after the method's class allows for parameter details to be overridden
             * by annotations in the subclass.
             */
            readParameters(resourceClass, null, true);
        }

        parameters.setPathItemParameters(getParameters(resourceMethod));
    }

    private void readConfigurationPathParameter(ClassInfo resourceClass) {
        FrameworkParameter frameworkParam = JaxRsParameter.PATH_PARAM.parameter;
        String path = scannerContext.getConfig().getScanResourceClasses()
                .get(OpenApiConstants.SCAN_RESOURCE_CLASS_PREFIX + resourceClass.name().toString());
        if (path != null) {
            readParameter(
                    new ParameterContextKey(path, frameworkParam.location, frameworkParam.defaultStyle),
                    null,
                    frameworkParam,
                    path,
                    resourceClass,
                    false);
        }
    }
```

Reply:
I think you'll need to change `pathOf` [1] to get the new configuration value in the `CLASS` case of the switch.

[1] https://github.com/smallrye/smallrye-open-api/blob/2cd874153e0d48f9946ed6d4190b6488cf53c59e/extension-jaxrs/src/main/java/io/smallrye/openapi/jaxrs/JaxRsParameterProcessor.java#L236-L245

Reply:
Thanks, it seems to work. Which do you think should take precedence, the path annotation or the configuration property if both are present?

Reply:
I think overriding the annotation with the configuration would make sense.

Reply:
Agree. I've pushed a working solution, just need to add some tests.

Reply:
I wanted to add tests today, but to be honest I'm not quite sure where and how it should be tested since I didn't find any tests for the existing options. In addition, sonarcloud complains about uncovered lines in `JaxRsAnnotationScanner`, but as far as I can see this class is not covered by tests at all, not just the lines that I changed / added.

Reply:
@anarsultanov the tests are generally code to use the main scanner entry point `OpenApiAnnotationScanner`. See the tests in `io.smallrye.openapi.runtime.scanner` in the jaxrs module's test directory.

You could add a new test class there and follow the pattern from the others to set your desired configuration, call the `scan` method, and compare the result against some known static JSON file. The class should extend `IndexScannerTestBase` where you can find methods to create simple implementations of the `OpenApiConfig` interface, for example `dynamicConfig(String key, Object value)`.

Reply:
Added some simple tests, please let me know if you think more cases need to be tested.